frontend/packages/ceph-storage-plugin/src/components/create-storage-system/reducer.ts

```diff
@@ -125,12 +121,80 @@ export type LocalVolumeSet = {
   chartNodes: Set<string>;
 };
 
+const setDeployment = (
+  prevState: WizardState,
+  newState: WizardState,
+  deploymentType: DeploymentType,
+) => {
+  /*
+   * Wizard state should be reset when a new deployment type is selected
+   * in order to avoid stale state collisions since each deployment mode
+   * has its own supported configuration.
+   *
+   * Its not required if the user has not visited more than first step.
+   */
+  if (prevState.stepIdReached !== 1) {
+    const { type } = prevState.backingStorage;
+    return {
+      ...initialState,
+      storageClass:
+        type === BackingStorageType.EXISTING ? prevState.storageClass : initialState.storageClass,
+      backingStorage: {
+        ...prevState.backingStorage,
+        deployment: deploymentType,
+      },
+    };
+  }
+
+  newState.backingStorage.deployment = deploymentType;
+  return newState;
+};
+
+const setBackingStorageType = (
+  prevState: WizardState,
+  newState: WizardState,
+  bsType: BackingStorageType,
+) => {
+  /*
+   * Wizard state should be reset when a new backing storage type is selected
+   * in order to avoid stale state collisions since each backing storage type
+   * has its own supported variables. e.g if arbiter was selected and not
+   * deselected before changing the backing storage type then storage cluster spec
+   * will mark the arbiter option as enabled.
+   *
+   * Its not required if the user has not visited more than first step.
+   */
+  if (prevState.stepIdReached !== 1) {
+    return {
+      ...initialState,
+      backingStorage: {
+        ...prevState.backingStorage,
+        type: bsType,
+      },
+    };
+  }
+
+  /* Update storage class state when existing storage class is not selected. */
+  if (bsType === BackingStorageType.LOCAL_DEVICES || bsType === BackingStorageType.EXTERNAL) {
+    newState.storageClass = {
+      name: '',
+      provisioner: bsType === BackingStorageType.EXTERNAL ? '' : NO_PROVISIONER,
+    };
+  }
+
+  /* Reset external storage when deselected. */
+  if (bsType !== BackingStorageType.EXTERNAL) {
+    newState.backingStorage.externalStorage = initialState.backingStorage.externalStorage;
+  }
+
+  newState.backingStorage.type = bsType;
+  return newState;
+};
+
 /* Reducer of CreateStorageSystem */
 export const reducer: WizardReducer = (prevState, action) => {
-  const newState = _.cloneDeep(prevState);
+  const newState: WizardState = _.cloneDeep(prevState);
```

Comment:
Since you're adding types. Please add types to the Line 195 as well.

Reply:
L195 has types `export const reducer: WizardReducer` :D

Reply:
sorry my bad